dashboard/src/app/utils/struct.ts

```diff
@@ -24,3 +24,26 @@ const structToJSONObject = (struct: Struct): Object => {
 export const structFromJSONString = (jsonStr: string): Struct => {
   return Struct.fromJSON(jsonStr)
 }
+
+export const getAttemptOutput = (output: VariableValue | undefined): string => {
+  if (!output?.value?.value) return 'No Output'
+  return String(getVariableValue(output))
+}
+
+export const getAttemptResult = (
+  result:
+    | { $case: 'output'; value: VariableValue }
+    | { $case: 'error'; value: any }
+    | { $case: 'exception'; value: any }
+    | undefined
+): string => {
+  if (!result) return 'No Output'
+
+  if (result.$case === 'output') {
+    return getAttemptOutput(result.value)
+  }
+
+  // For error/exception, prefer a message property if present, otherwise fallback to JSON
+  const val = result.value
+  return (val && (val.message ?? val.msg ?? val.toString())) || JSON.stringify(val) || 'No Output'
+}
```

Comment:
What is this? you should use the types from the littlehorse-client library. Also avoid using `any`